src/preact.d.ts

```diff
@@ -277,8 +279,8 @@ declare namespace JSX {
 		charSet?:string;
 		challenge?:string;
 		checked?:boolean;
-		class?:string;
-		className?:string;
+		class?:string | Object;
+		className?:string | Object;
```

Comment:
read your example again, this should be then: `string | {[key:string]:string|boolean}`